projects/telemetry/src/nmea_checksum.c

```diff
@@ -37,12 +39,11 @@ bool nmea_compare_checksum(char *message, size_t message_len) {
   }
 
   // Extracts the received checksum
-  uint32_t received_checksum = 0;
-  sscanf(received, "%2x", (unsigned int *)&received_checksum);
+  uint32_t received_checksum = (uint32_t)strtol(received, NULL, 16);
```

Comment:
If you're going to use `strtol`, why not use `strtoul`? Alternatively, you could write a simple hex to int function. Since your checksum should always be 2 hex characters, your checksum should always be a `uint8_t`.